common/flatpak-utils.c

```diff
@@ -5854,3 +5856,47 @@ flatpak_levenshtein_distance (const char *s, const char *t)
 
   return dist (s, ls, t, lt, 0, 0, d);
 }
+
+void
+flatpak_get_window_size (int *rows, int *cols)
+{
+  struct winsize w;
+
+  if (ioctl (STDOUT_FILENO, TIOCGWINSZ, &w) == 0)
+    {
+      *rows = w.ws_row;
+      *cols = w.ws_col;
+    }
+  else
+    {
+      *rows = 24;
+      *cols = 80;
+    }
+}
+
+void
+flatpak_get_cursor_pos (int* row, int *col)
+{
+  fd_set readset;
+  struct timeval time;
+  struct termios term, initial_term;
+
+  tcgetattr (STDIN_FILENO, &initial_term);
+  term = initial_term;
+  term.c_lflag &= ~ICANON;
+  term.c_lflag &= ~ECHO;
+  tcsetattr (STDIN_FILENO, TCSANOW, &term);
+
+  printf ("\033[6n");
+  fflush (stdout);
+
+  FD_ZERO (&readset);
+  FD_SET (STDIN_FILENO, &readset);
+  time.tv_sec = 0;
+  time.tv_usec = 100000;
+
+  if (select (STDIN_FILENO + 1, &readset, NULL, NULL, &time) == 1)
+    scanf ("\033[%d;%dR", row, col);
```

Comment:
I think we need to handle errors here, in case e.g. something else was in the stdin buffer already or something.